platform/ui-next/src/contextProviders/ViewportActionCornersProvider.tsx

```diff
@@ -0,0 +1,546 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useReducer,
+  ReactNode,
+  useMemo,
+} from 'react';
+import { ViewportActionCornersLocations } from '../components/Viewport/ViewportActionCorners';
+import { ActionComponentInfo, AlignAndSide } from '../types';
+
+// Define the default state
+const DEFAULT_STATE = {
+  viewports: {} as Record<string, Record<ViewportActionCornersLocations, ActionComponentInfo[]>>,
+};
+
+interface ViewportActionCornersApi {
```

Comment:
This interface is embedding knowledge about what the corners api  content can do into the corners api itself.  Do you actually need it, or can the individual content items have their own calls/settings to the API?

Reply:
Hmm, maybe these are just about the state of the items themselves, and not the internals.   Maybe add a comment about the api stating that it affects the corner item itself, and isnt for affecting external things.

Reply:
Added comment